Automattic/wp-calypso
# Reader Recent Overhaul: Add onboarding

Related to https://github.com/Automattic/loop/issues/201

## Proposed Changes

* This PR displays the Reader Onboarding component and an intro message if the user has no recent feed data.
* It introduces a `forceShow` prop on the `ReaderOnboarding` component so that it ignores the component's display criteria and forces it to show in this empty recent feeds context.
* It refreshes the recent feed data when the user clicks the final "Continue" button in Reader Onboarding. This closes Reader Onboarding and shows the user their new recent feed.
* It also adds a check to the Recent component that checks if the recent feed data has loaded for the first time before rendering any content. This prevents the flashing of components that are not in context while data is loading.

Demo:


https://github.com/user-attachments/assets/b0f255bb-c4a9-48d0-88bd-14388a5addac

## Why are these changes being made?



* Adds onboarding for users who are not yet following any blogs.

## Testing Instructions



* With an account that is not following any blogs
* Use Calypso Live and go to /read?flags=reader/recent-feed-overhaul
* Interact with reader onboarding and try to break it
* Once you're following a blog, reader onboarding should not be visible on /read?flags=reader/recent-feed-overhaul

## Pre-merge Checklist
